src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -550,7 +550,7 @@ ReactDOMComponent.Mixin = {
    * @return {string} Markup of opening tag.
    */
   _createOpenTagMarkupAndPutListeners: function(transaction, props) {
-    var ret = '<' + this._tag;
+    var ret = '<' + this._currentElement.type;
```

Comment:
We need this to line up with https://github.com/facebook/react/blob/master/src/renderers/dom/shared/ReactDOMComponent.js#L536